plugins/rest/aws.go

```diff
@@ -186,6 +209,40 @@ func (cs *awsMetadataCredentialService) credentials() (awsCredentials, error) {
 	return cs.creds, nil
 }
 
+func isECS() bool {
+	// the special relative path URI is set by the container agent in the ECS environment only
+	_, isECS := os.LookupEnv(ecsRelativePathEnvVar)
+	return isECS
+}
+
+func doMetaDataRequestWithClient(req *http.Request, client *http.Client, desc string) ([]byte, error) {
+	// convenience function to get the body of an AWS EC2 metadata service request with
+	// appropriate error-handling boilerplate and logging for this special case
+	resp, err := client.Do(req)
+	if err != nil {
+		// some kind of catastrophe talking to the EC2 service
+		return nil, errors.New(desc + " HTTP request failed: " + err.Error())
+	}
+	defer resp.Body.Close()
+
+	logrus.WithFields(logrus.Fields{
+		"url":     req.URL.String(),
+		"status":  resp.Status,
+		"headers": resp.Header,
+	}).Debug("Received response from " + desc + " service.")
```

Comment:
Not 100% sure what all is included, but I'm usually hesitant to dump headers in the log, is there any possibility that a token might be in there?

Reply:
Yes, there probably could be (although, in this case, only fairly short-lived ones). I was following the pattern that was there in the rest package when I first found it:

https://github.com/open-policy-agent/opa/blob/ea20b6a222b54132c79fdf080432bd778d04726c/plugins/rest/rest.go#L215-L219

https://github.com/open-policy-agent/opa/blob/ea20b6a222b54132c79fdf080432bd778d04726c/plugins/rest/rest.go#L223-L230

If we are concerned about sensitive values getting logged, I would imagine the code in the rest package would be a significant higher risk for e.g. bearer token authentication. That said, I'd rather keep the PR about the IMDSv2 implementation and we could have a separate one to sanitize/normalize the rest logging code.

Reply:
No worries, if we are already doing that then lets keep it as-is. I guess the logic is that since they're at debug log levels its OK.